app/assets/javascripts/modules/track-click.js

```diff
@@ -10,14 +10,24 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       var options = {},
           category = element.attr('data-track-category'),
           action = element.attr('data-track-action'),
-          label = element.attr('data-track-label');
+          label = element.attr('data-track-label'),
+          dimension = element.attr('data-track-dimension'),
+          customDimension = element.data('track-custom-dimension');
 
       if (label) {
         options.label = label;
       }
 
       function trackClick() {
-        if (GOVUK.analytics && GOVUK.analytics.trackEvent) {
+        if (!GOVUK.analytics) {
+          return;
+        }
+
+        if (customDimension && dimension && GOVUK.analytics.setDimension) {
+          GOVUK.analytics.setDimension(customDimension, dimension);
```

Comment:
I think this dimension will remain set for all subsequent tracked-clicks.

eg If I cmd+click and open a page in a new tab. Then on the old page if I click an external link or a download – (two types of links that get tracked), those will be tracked with this dimension.